hdf5-types/src/dyn_value.rs

```diff
@@ -703,6 +703,7 @@ impl<'a> DynValue<'a> {
             FixedUnicode(_) => DynFixedString::new(buf, true).into(),
             VarLenAscii => DynVarLenString::new(buf, false).into(),
             VarLenUnicode => DynVarLenString::new(buf, true).into(),
+            Reference(_x) => todo!(),
```

Comment:
```suggestion
            Reference(_x) => todo!("References are not yet supported as fill value"),
```

Reply:
Since we can't use `h5call` and family in `hdf5-types` we will struggle to support this feature without some heavy refactoring.